dubbo-registry/dubbo-registry-default/src/test/java/org/apache/dubbo/registry/dubbo/RegistryDirectoryTest.java

```diff
@@ -1014,6 +1014,37 @@ public void test_Notified_acceptProtocol2() {
         List<Invoker<DemoService>> invokers = registryDirectory.list(invocation);
         Assert.assertEquals(2, invokers.size());
     }
+    
+    @Test
+    public void test_Notified_withGroupFilter() {
+    	URL directoryUrl = noMeaningUrl.addParameterAndEncoded(Constants.REFER_KEY, "interface" + service + "&group=group1,group2");
+    	RegistryDirectory directory = this.getRegistryDirectory(directoryUrl);
+    	URL provider1 = URL.valueOf("dubbo://10.134.108.1:20880?methods=getXXX&group=group1&mock=false");
+    	URL provider2 = URL.valueOf("dubbo://10.134.108.1:20880?methods=getXXX&group=group2&mock=false");
+    	
+    	List<URL> providers = new ArrayList<>();
+    	providers.add(provider1);
+    	providers.add(provider2);
+    	directory.notify(providers); 
+    	
+    	invocation = new RpcInvocation();
+    	invocation.setMethodName("getXXX");
+    	List<Invoker<DemoService>> invokers = directory.list(invocation);
+    	
+    	Assert.assertEquals(2, invokers.size());
+    	Assert.assertTrue(invokers.get(0) instanceof MockClusterInvoker);
+    	Assert.assertTrue(invokers.get(1) instanceof MockClusterInvoker);
+    	
+    	directoryUrl = noMeaningUrl.addParameterAndEncoded(Constants.REFER_KEY, "interface" + service + "&group=group1");

```

Comment:
group1 can be also defined a variable so that it can be reuse here, so in future even we change value of it, it would be a single place change only.

Reply:
I appreciate your spirit of code refinement, we should code with specification,and I also found that some unit tests were not written in strict accordance with the code specification，I wonder if we have any hard and fast rules for coding unit tests @carryxyh

Reply:
@lixiaojiee Would you please bring your question to the dev mailing list, we can discuss it there. For now, it looks good to me. I will merge it.